src/functTest/groovy/com/bmuschko/gradle/docker/DockerWorkflowFunctionalTest.groovy

```diff
@@ -494,6 +494,70 @@ class DockerWorkflowFunctionalTest extends AbstractFunctionalTest {
         result.output.contains("Devices : [/dev/sda:/dev/xvda:rwm]")
     }
 
+    def "Can build an image, create a container and assign a network and alias"() {
+        File imageDir = temporaryFolder.newFolder('images', 'minimal')
+        File dockerFile = createDockerfile(imageDir)
+
+        String uniqueContainerName = createUniqueContainerName()
+        String uniqueNetworkName = createUniqueNetworkName()
+
+        buildFile << """
+            import com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask
+            import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+            import com.bmuschko.gradle.docker.tasks.container.DockerCreateContainer
+            import com.bmuschko.gradle.docker.tasks.container.DockerInspectContainer
+
+            class DockerCreateNetworkTask extends AbstractDockerRemoteApiTask {
+                @Override
+                void runRemoteCommand(dockerClient) {
+                    dockerClient.createNetworkCmd().withName("${uniqueNetworkName}").exec()
+                }
+            }
+            class DockerRemoveNetworkTask extends AbstractDockerRemoteApiTask {
+                @Override
+                void runRemoteCommand(dockerClient) {
+                    dockerClient.removeNetworkCmd("${uniqueNetworkName}").exec()
+                }
+            }
+
+            task removeNetwork(type: DockerRemoveNetworkTask)
+            task createNetwork(type: DockerCreateNetworkTask)
+
+            task buildImage(type: DockerBuildImage) {
+                inputDir = file("${dockerFile.parentFile.path}")
+                tag = "${createUniqueImageId()}"
+            }
+
+            task createContainer(type: DockerCreateContainer) {
+                dependsOn buildImage, createNetwork
+                targetImageId { buildImage.getImageId() }
+                containerName = "${uniqueContainerName}"
+                networkMode = "${uniqueNetworkName}"
+                networkAliases = ["some-alias"]
+                cmd = ['/bin/sh']
+            }
+
+            removeNetwork.mustRunAfter createNetwork
+            createNetwork.finalizedBy removeNetwork
```

Comment:
This is your problem. You're tearing down the network just after you create it! :)

Do something like `inspectContainer.finalizedBy removeNetwork` and you should be good to go. Just ran this setup on my linux machine and it worked as expected.